packages/react/menu/src/Menu.tsx

```diff
@@ -661,15 +665,28 @@ const MenuItem = React.forwardRef((props, forwardedRef) => {
       {...itemProps}
       ref={composedRefs}
       disabled={disabled}
-      // we handle selection on `pointerUp` rather than `click` to match native menus implementation
-      onPointerUp={composeEventHandlers(props.onPointerUp, handleSelect)}
+      onClick={composeEventHandlers(props.onClick, handleSelect)}
+      onPointerDown={(event) => {
+        props.onPointerDown?.(event);
+        isPointerDownRef.current = true;
+      }}
+      onPointerUp={composeEventHandlers(props.onPointerUp, (event) => {
+        // Pointer down can move to a different menu item which should activate it on pointer up.
+        // We dispatch a click for selection to allow composition with click based triggers.
+        if (!isPointerDownRef.current) event.currentTarget?.click();
```

Comment:
I find it a bit odd that if users add an `onClick` listener to the item, it will trigger even when it's not a click (down and up), I'm guessing that was the only way we managed to get it to work with `Dialog` trigger?

Reply:
> I'm guessing that was the only way we managed to get it to work with Dialog trigger?

Yes and I've also realised this conveniently fixes #745 too.

I'm okay with it firing a click tbh because that's what we're logically determining to be a click on the item - the circumstances in which it would activate. I guess we can see what complaints we get here from consumers, if any.